src/dotnet/src/Mil.Navy.Nrl.Norm/NormNode.cs

```diff
@@ -0,0 +1,164 @@
+        /// </summary>
+        public const long NORM_NODE_ANY = 0xffffffff;
+        /// <summary>
+        /// The special value NORM_NODE_NONE corresponds to an invalid (or null) node.
+        /// </summary>
+        public const int NORM_NODE_NONE = 0;
+        /// <summary>
+        /// The special value NORM_NODE_INVALID corresponds to an invalid reference.
+        /// </summary>
+        public const int NORM_NODE_INVALID = 0;
+        /// <summary>
+        /// The handle is associated to the NORM protocol engine instance.
+        /// </summary>
+        private long _handle;
+
+        /// <summary>
+        /// Parameterized contructor.
+        /// </summary>
+        /// <param name="handle">The handle is associated to the NORM protocol engine instance.</param>
+        internal NormNode(long handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// The NormNodeId identifier for the remote participant.
+        /// </summary>
+        public long Id => NormNodeGetId(_handle);
+
+        /// <summary>
+        /// The current network source address detected for packets received from remote NORM sender.
+        /// </summary>
+        public unsafe IPEndPoint Address
+        {
+            get
+            {
+                var bufferLength = 256;
+                var buffer = stackalloc byte[bufferLength];
+                var addrBuffer = (nint)buffer;
+
+                if (!NormNodeGetAddress(_handle, addrBuffer, ref bufferLength, out int port))
+                {
+                    throw new IOException("Failed to get node address");
+                }
+
+                var addressBytes = new ReadOnlySpan<byte>(buffer, bufferLength);
+                var ipAddress = new IPAddress(addressBytes);
+                return new IPEndPoint(ipAddress, port);
+            }
+        }
+
+        /// <summary>
+        /// The advertised estimate of group round-trip timing (GRTT) for the remote sender.
+        /// </summary>
+        public double Grtt => NormNodeGetGrtt(_handle);
+
+        /// <summary>
+        /// This function retrieves the content of an application-defined command that was received from a remote sender.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the offset or length are outside of the buffer.</exception>
+        public int GetCommand(byte[] buffer, int offset, int length)
+        {
+            if (offset < 0 || offset >= buffer.Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(offset), "The offset is out of range");
+            }
+            if (length < 1 || offset + length > buffer.Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(length), "The length is out of range");
+            }
+
+            var bufferHandle = GCHandle.Alloc(buffer, GCHandleType.Pinned);
```

Comment:
I think a nicer way to write this is using a [fixed](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/fixed) statement and `byte*` in the PInvoke signature.